src/modules/endpoint/sagas/watchFetchEvent.js

```diff
@@ -5,31 +5,40 @@ import { all, call, put, takeLatest } from 'redux-saga/effects'
 import type { FetchEventActionType, FetchEventFailedActionType, PushEventActionType } from '../../app/StoreActionType'
 import type { DataContainer } from '../DataContainer'
 import loadCityContent from './loadCityContent'
+import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import DatabaseContext from '../DatabaseContext'
 
 function * fetchEvent (dataContainer: DataContainer, action: FetchEventActionType): Saga<void> {
-  const {city, language, path, key, forceUpdate, shouldRefreshResources} = action.params
+  const {city, language, path, key, criterion} = action.params
   try {
-    yield call(loadCityContent, dataContainer, city, language, forceUpdate, shouldRefreshResources)
+    const loadCriterion = new ContentLoadCriterion(criterion)
+    const allContentLoaded = yield call(loadCityContent,
+      dataContainer, city, language,
+      loadCriterion
+    )
+    if (allContentLoaded) {
+      const context = new DatabaseContext(city, language)
+      const [events, resourceCache, languages] = yield all([
+        call(dataContainer.getEvents, context),
+        call(dataContainer.getResourceCache, context),
+        call(dataContainer.getLanguages, context)
+      ])
 
-    const [events, resourceCache, languages] = yield all([
-      call(dataContainer.getEvents),
-      call(dataContainer.getResourceCache),
-      call(dataContainer.getLanguages)
-    ])
-
-    const insert: PushEventActionType = {
-      type: `PUSH_EVENT`,
-      params: {
-        events,
-        resourceCache,
-        path,
-        languages,
-        key,
-        city,
-        language
+      const insert: PushEventActionType = {
+        type: `PUSH_EVENT`,
+        params: {
+          events,
+          resourceCache,
+          path,
+          languages,
+          key,
+          city,
+          language,
+          peek: loadCriterion.peek()
+        }
       }
+      yield put(insert)
     }
```

Comment:
`else` do nothing? How does the app react here?

Reply:
If `allContentLoaded` is false we are not able to put events, categories etc. in the state.

This usually happens because the language was not available. In that case it is needed to trigger a refetch

Reply:
So no loading for an infinite time anymore? I had that problem when removing the `throw Error`